CTP/shell/src/com/navercorp/cubridqa/shell/deploy/DeployHA.java

```diff
@@ -0,0 +1,83 @@
+		
+		port = context.getProperty("env." + masterEnvId + ".ssh.port");
+		user = context.getProperty("env." + masterEnvId + ".ssh.user");
+		pwd = context.getProperty("env." + masterEnvId + ".ssh.pwd");
+		masterHost = context.getProperty("env." + masterEnvId + ".ssh.host");
+		
+		envIdentify = "MasterEnvId=" + masterEnvId + "[" + user+"@"+ masterHost +":" + port + "] - SlaveEnvId:" + slaveIp;
+		this.ssh = new SSHConnect(masterHost, port, user, pwd, context.getServiceProtocolType());
+		this.log = log;
+	}
+	
+	
+	public void deploy()
+	{
+		log.print("===== Start to update HA.properties =====");
+		ShellInput scripts = new ShellInput();
+		scripts.addCommand("ini.sh -u MASTER_SERVER_IP=" + masterHost + " $init_path/HA.properties");
+		scripts.addCommand("ini.sh -u MASTER_SERVER_USER=" + this.user + " $init_path/HA.properties");
+		scripts.addCommand("ini.sh -u MASTER_SERVER_PW=" + this.pwd + " $init_path/HA.properties ");
+		scripts.addCommand("ini.sh -u MASTER_SERVER_SSH_PORT=" + this.port + " $init_path/HA.properties");
+		
+		scripts.addCommand("ini.sh -u SLAVE_SERVER_IP=" + slaveIp + " $init_path/HA.properties");
+		scripts.addCommand("ini.sh -u SLAVE_SERVER_USER=" + this.user + " $init_path/HA.properties");
+		scripts.addCommand("ini.sh -u SLAVE_SERVER_PW=" + this.pwd + " $init_path/HA.properties ");
+		scripts.addCommand("ini.sh -u SLAVE_SERVER_SSH_PORT=" + this.port + " $init_path/HA.properties");
+		
+		scripts.addCommand("cubrid_port_id_value=`ini.sh -s 'common' $CUBRID/conf/cubrid.conf cubrid_port_id`");
+		scripts.addCommand("ini.sh -u CUBRID_PORT_ID=$cubrid_port_id_value $init_path/HA.properties");
+		scripts.addCommand("cubrid_master_shm_id_value=`ini.sh -s 'broker' $CUBRID/conf/cubrid_broker.conf MASTER_SHM_ID`");
+		scripts.addCommand("ini.sh -u MASTER_SHM_ID=$cubrid_master_shm_id_value $init_path/HA.properties");
+		scripts.addCommand("cubrid_broker1_port_value=`ini.sh -s '%query_editor' $CUBRID/conf/cubrid_broker.conf BROKER_PORT` ");
+		scripts.addCommand("ini.sh -u BROKER_PORT1=$cubrid_broker1_port_value $init_path/HA.properties ");
+		scripts.addCommand("cubrid_broker1_app_server_shm_value=`ini.sh -s '%query_editor' $CUBRID/conf/cubrid_broker.conf APPL_SERVER_SHM_ID`");
+		scripts.addCommand("ini.sh -u APPL_SERVER_SHM_ID1=$cubrid_broker1_app_server_shm_value $init_path/HA.properties ");
+		scripts.addCommand("cubrid_broker2_port_value=`ini.sh -s '%BROKER1' $CUBRID/conf/cubrid_broker.conf BROKER_PORT` ");
+		scripts.addCommand("ini.sh -u BROKER_PORT2=$cubrid_broker2_port_value $init_path/HA.properties ");
+		scripts.addCommand("cubrid_broker2_app_server_shm_value=`ini.sh -s '%BROKER1' $CUBRID/conf/cubrid_broker.conf APPL_SERVER_SHM_ID`");
+		scripts.addCommand("ini.sh -u APPL_SERVER_SHM_ID2=$cubrid_broker2_app_server_shm_value $init_path/HA.properties");
+		scripts.addCommand("ini.sh -u configPath=$init_path $init_path/make_ha.sh ");
+		
+		String haPortId = context.getInstanceProperty(this.masterEnvId, "ha.ha_port_id"); 
```

Comment:
It's better to use default port for ha_port_id if it is not provided.

if(haPortId == null || haPortId.trim().equals("")) {
    haPortId  = "59901";
}